src/java/org/apache/cassandra/service/accord/AccordKeyspace.java

```diff
@@ -595,13 +664,230 @@ public static UntypedResultSet loadCommandRow(CommandStore commandStore, TxnId t
     {
         String cql = "SELECT * FROM %s.%s " +
                      "WHERE store_id = ? " +
+                     "AND domain = ? " +
                      "AND txn_id=(?, ?, ?)";
 
         return executeInternal(String.format(cql, ACCORD_KEYSPACE_NAME, COMMANDS),
                                commandStore.id(),
+                               txnId.domain().ordinal(),
                                txnId.msb, txnId.lsb, txnId.node.id);
     }
 
+    public static void findAllCommandsByDomain(int commandStore, Routable.Domain domain, Set<String> columns, Observable<UntypedResultSet.Row> callback)
+    {
+        WalkCommandsForDomain work = new WalkCommandsForDomain(commandStore, domain, columns, Stage.READ.executor(), callback);
+        work.schedule();
+    }
+
+    private static abstract class TableWalk implements Runnable, DebuggableTask
```

Comment:
I deleted most of my comments on this CQL stuff since it seems temporary anyways.